src/lib.rs

```diff
@@ -311,3 +346,82 @@ impl Write for Device {
         Ok(())
     }
 }
+
+impl Device {
+    pub fn set_mpsse_clock(&mut self, freq: u32) -> std::result::Result<(), io::Error> {
+        const MAX: u32 = 30_000_000;
+        const MIN: u32 = 92;
+
+        assert!(
+            freq >= MIN,
+            "frequency of {} exceeds minimum of {}",
+            freq,
+            MIN
+        );
+        assert!(
+            freq <= MAX,
+            "frequency of {} exceeds maximum of {}",
+            freq,
+            MAX
+        );
+
+        let (divisor, clkdiv) = if freq <= 6_000_000 {
+            (6_000_000 / freq - 1, Some(true))
+        } else {
+            (30_000_000 / freq - 1, Some(false))
+        };
+
+        self.write_all(MpsseCmdBuilder::new().set_clock(divisor, clkdiv).as_slice())?;
+
+        Ok(())
+    }
+}
+
+impl MpsseCmdExecutor for Device {
+    type Error = io::Error;
+
+    fn init(&mut self, settings: &MpsseSettings) -> std::result::Result<(), io::Error> {
+        let millis = u8::try_from(settings.latency_timer.as_millis())
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
```

Comment:
I'm not sure here, maybe it could make sense to just use the maximum value if the user requests a longer one... I don't know whether `MpsseCmdExecutor::init` is supposed to be exact in this regard or best effort.
On the other side, a user can pass a Duration of, say, 0.25 ms, which will become 0 here. IIRC, `libftdi` checks for that and reports an error in `set_latency_timer`.

Reply:
I would prefer to get explicit 'latency value out of range' error rather than silent truncation to the maximum value. But I am open to any suggestions here.

Reply:
Okay, let's leave this as is.